app/src/main/java/com/fcaputo/parkingapp/mvp/presenter/SettingsPresenter.kt

```diff
@@ -0,0 +1,33 @@
+package com.fcaputo.parkingapp.mvp.presenter
+
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+
+class SettingsPresenter(private val model: SettingsContract.Model, private val view: SettingsContract.View) : SettingsContract.Presenter {
+    init {
+        view.onSaveButton { onSaveButton() }
+        view.showParkingSize(getSizeString(model.getParkingSize()))
+    }
+
+    override fun onSaveButton() {
+        if(formIsIncomplete()) {
+            view.showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            return
```

Comment:
What is the purpose of this return?

Reply:
To avoid nesting and nesting within if-else branches. Otherwise, the rest of the code should be in the other branch, and for every other validation, it would imply another if-else with more nesting